src/main/java/net/nuagenetworks/bambou/ssl/DynamicKeystoreGenerator.java

```diff
@@ -0,0 +1,156 @@
+
+			RSAPrivateKey rsaPrivateKey = generateRSAPrivateKey(decodedPrivateKey);
+
+			KeyStore keystore = createAndLoadDynamicKeystore(rsaPrivateKey, certificate);
+			keyManagers = generateKeyManagersForDynamicKeystore(keystore);
+			return keyManagers;
+		} catch (Exception e) {
+			throw new KeyManagementException(e);
+		}
+	}
+
+	public static Certificate getCertificate(String certificatePEMFile) throws KeyManagementException {
+        logger.debug("Creating a Certificate for file : " + certificatePEMFile);
+		Certificate certificate = null;
+		try (FileInputStream fis = new FileInputStream(certificatePEMFile)) {
+			certificate = generateX509Certificate(fis);
+		} catch (IOException e) {
+			throw new KeyManagementException(e);
+		}
+	    return certificate;
+	}
+	
+	public static Certificate generateX509Certificate(InputStream is) throws KeyManagementException {
+        logger.debug("Generating X509 Certificate");
+		Certificate certificate = null;
+		try {
+			CertificateFactory cf = CertificateFactory.getInstance(CERTIFICATE_FACTORY_INSTANCE_TYPE);
+			certificate = cf.generateCertificate(is);
+		} catch (CertificateException e) {
+			throw new KeyManagementException(e);
+		}
+	    return certificate;
+	}
+	
+	public static String getContentsOfPrivateKeyPEMFile(String privateKeyPEMFile) throws KeyManagementException {
+        logger.debug("Reading key contents for file : " + privateKeyPEMFile);
+	    byte[] keyFileBytes = {};
+	    String privateKeyContents = "";
+		try {
+			Path path = Paths.get(privateKeyPEMFile);
+			keyFileBytes = Files.readAllBytes(path);
+			privateKeyContents = new String(keyFileBytes, KEY_CHARSET);
+		} catch (IOException e) {
+			throw new KeyManagementException(e);
+		}
+		return privateKeyContents;
+	}
+	
+	public static byte[] getDecodedPrivateKey(String keyFileContents) {
+        logger.debug("Decoding private key : " + keyFileContents);
+	    byte[] decodedPrivateKey = {};
+		Base64 base64Decoder = new Base64();
+		String privateKey = keyFileContents.replaceAll(PRIVATE_KEY_HEADER_TRAILER_REGEX, "");
+		decodedPrivateKey = base64Decoder.decode(privateKey);
+		return decodedPrivateKey;
+	}
+	
+	public static RSAPrivateKey generateRSAPrivateKey(byte[] decodedPrivateKey) throws KeyManagementException {
+        logger.debug("Generating RSA private key");
+		PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(decodedPrivateKey);
+		KeyFactory keyFactory;
+		RSAPrivateKey rsaPrivateKey = null;
+		try {
+			keyFactory = KeyFactory.getInstance(KEY_FACTORY_INSTANCE_TYPE);
+			rsaPrivateKey = (RSAPrivateKey) keyFactory.generatePrivate(spec);
+		} catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+			throw new KeyManagementException(e);
+		}
+		return rsaPrivateKey;
+	}
+	
+	public static KeyStore createAndLoadDynamicKeystore(RSAPrivateKey rsaPrivateKey, Certificate certificate) throws KeyManagementException {
+        logger.debug("Generating Keystore from RSA private key and X509 certificate");
```

Comment:
Indentation issue.  Make sure to use spaces instead of tabs. Applies to all changes.